### PR TITLE
feat: add masking for captured data

### PR DESCRIPTION
**Pull request by TristanSpeakEasy**
dependant on:
* https://github.com/speakeasy-api/speakeasy-schemas/pull/12
* https://github.com/speakeasy-api/sdk-test-suite/pull/3

awaiting decision on: https://speakeasyapi.slack.com/archives/C036B83TW3X/p1660737645161639

### CODE REVIEW

**Comment by TristanSpeakEasy** [comment]
The tests/lint and building will fail until we merge https://github.com/speakeasy-api/speakeasy-schemas/pull/12

File: README.md [hunk 1]
```diff
@@ -131,4 +143,53 @@ func MyHandler(w http.ResponseWriter, r *http.Request) {
 }
 ```
 
-Note: This is not required, but is highly recommended. By setting a customer ID you can easily associate requests with your customers/users in the Speakeasy Dashboard, powering filters in the Request Viewer [(Coming soon)](https://docs.speakeasyapi.dev/speakeasy-user-guide/request-viewer-coming-soon).
\ No newline at end of file
+Note: This is not required, but is highly recommended. By setting a customer ID you can easily associate requests with your customers/users in the Speakeasy Dashboard, powering filters in the Request Viewer [(Coming soon)](https://docs.speakeasyapi.dev/speakeasy-user-guide/request-viewer-coming-soon).
```
Comment (alexadrake) [inline README.md:146]: I updated the docs, if you want to change the `(Coming soon)` to something else

File: internal/bodymasking/bodymasking.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	numberFieldMatchRegex = `("%s": *)(-?[0-9]+\.?[0-9]*)( *[, \n\r}]?)`
+)
+
+func MaskBodyRegex(body, mimeType string, stringMasks map[string]string, numberMasks map[string]string) (string, error) {
+	mediaType, _, err := mime.ParseMediaType(mimeType)
+	if err != nil {
+		return "", err
+	}
+
+	if mediaType != "application/json" {
```
Comment (alexadrake) [inline internal/bodymasking/bodymasking.go:28]: Should there be a callout in the readme or something that this is only set to work with json for the time being?

**Review by alexadrake — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -131,4 +143,53 @@ func MyHandler(w http.ResponseWriter, r *http.Request) {
...
+
+## Masking sensitive data
+
+Speakeasy can mask sensitive data in the query string parameters, headers, cookies and request/response bodies captured by the SDK. This is useful for protecting against sensitive data leakage, and retaining control over the data that is captured.
```
Comment (simplesagar) [inline README.md:150]:
```suggestion
Speakeasy can mask sensitive data in the query string parameters, headers, cookies and request/response bodies captured by the SDK. This is useful for maintaining sensitive data isolation, and retaining control over the data that is captured.
```

**Review by simplesagar — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (TristanSpeakEasy) [reply README.md]: done

File: internal/bodymasking/bodymasking.go (hunk 2 above)
Comment (TristanSpeakEasy) [reply internal/bodymasking/bodymasking.go:28]: done

**Pull request merged**